objects/obj_boss_portal/Step_0.gml

```diff
@@ -0,0 +1,26 @@
+/// @description Player collision + button pressed
+if room != room_shop {
+	key_open_door = global.key_interact;
+}else {
+	key_open_door = global.key_interact;
+}
```

Comment:
There's no point in having this if statement if both blocks of code are identical.
```suggestion
key_open_door = global.key_interact;
```